tests/test_lab_dev/test_states.py

```diff
@@ -87,6 +87,16 @@ def test_bargmann_triple_error(self):
         with pytest.raises(AttributeError):
             Number([0], n=10).bargmann
 
+    def test_normalize(self):
+        # Bargmann
+        coh = Coherent([0], x=1.0, y=1.0) * 0.5
+        normalized = coh.normalize()
+        assert np.isclose(normalized.probability, 1.0)
+        # Fock
+        coh.to_fock(5)  # truncated
+        normalized = coh.normalize()
+        assert np.isclose(normalized.probability, 1.0)
+
```

Comment:
I was curious to check whether the batched case worked too, so in case you want, you could use this code.

```suggestion
    @pytest.mark.parametrize(
        "state",
        [
            0.5 * Coherent([0], x=1.0, y=1.0),
            0.3 * (Coherent([0], x=1.0, y=1.0) + Coherent([0], x=-1.0, y=-1.0)),
        ],
    )
    def test_normalize(self, state):
        # Bargmann
        normalized = state.normalize()
        assert np.isclose(normalized.probability, 1.0)
        # Fock
        state.to_fock(5)  # truncated
        normalized = state.normalize()
        assert np.isclose(normalized.probability, 1.0)
```

Reply:
thanks!